### ISSUE TITLE
Protect discard button even more!

### ISSUE THREAD

**Issue by MHM5000** [sublimehq/sublime_merge#1383]
**Problem description**
Discard/Discard All button is really dangerous!
We could prevent something terrible from happening.

**Preferred solution**
Password protecting the discard/discard all button + red color for the pop to indicate danger. in ubuntu, the popup's button is not red. just a border around the discard button.
Obviously, there could be a flag for this feature in the settings.

**Alternatives**
+ a recycle bin containing discarded changes. this can come in handy. it also could easily become a git feature. it is basically the stash functionality with an indicator that will show it is a deleted piece of data.
+ a timer to enable the confirm button in the pop-up; or a timer to enable the 2nd click of the button.

**Screenshots**
no

____
I almost lost 7 days' worth of work. let's make this thing a reality.

**Comment by sylbru** [comment]
Agreed, someone might even have a faulty mouse that occasionnally doubleclicks, and that Discard button shows no mercy :)

**Comment by infabo** [comment]
> Discard/Discard All button is really dangerous!

I agree for the "Discard" button near a single file. It needs a second click to confirm - but people with faulty mouse double-clicks can this be harm. At least this should be throttled. Second click within e.g. 50ms should not be accepted.

For the "Discard All" button:
This already opens a modal dialog!

![image](https://user-images.githubusercontent.com/3999104/147935981-017572f4-a197-4485-bcf1-2d4608e9c34c.png)

Arguing, not reading the warning message and actively hitting the DISCARD button -> I can see no reason to improve that here in any way. I can't see why this action should be password-protected. And even when -> what credentials to use? Merge is cross-platform and is a wrapper around the git-binary. No need to tinker with platform-specific keystores.

But one improvement I can see though: The focus is on the "DISCARD" button by default. It should be on the "CANCEL"-button instead. So people using keyboards not accidentally hit ENTER.

**Comment by sylbru** [comment]
Throttling is a great idea, as well as not giving the focus to Discard by default in that dialog.

**Comment by MHM5000** [comment]
I can see why you might say 
> I can't see why this action should be password-protected.
and If this suggestion gets discarded, I might create a package for it myself, assuming that can be done, but imagine if you had 20 files or a large edit not committed. one more barrier to stop you from losing money or time won't hurt.

**Comment by infabo** [comment]
I understand your concerns. I have lost changes in the past too. Mostly when using something like `git clean -fd`. I did not head of to Linus Torvalds complaining that I lost all my work. I just did not RTFM - as I could have used the dry-run option - or generally - checking my changes before making tabula rasa.

**Comment by jakobvase** [comment]
I would absolutely hate this. So please make it configurable. I want to be able to delete it all with just a double-click. I understand peoples' risk-willingness is different.
